RecoEgamma/EgammaTools/plugins/EGExtraInfoModifierFromDB.cc

```diff
@@ -555,79 +573,66 @@ void EGExtraInfoModifierFromDB::modifyObject(pat::Electron& ele) const {
   ele.correctMomentum(newMomentum, ele.trackMomentumError(), combinedMomentumError);
 }
 
-void EGExtraInfoModifierFromDB::modifyObject(pat::Photon& pho) const {
-  // we encounter two cases here, either we are running AOD -> MINIAOD
-  // and the value maps are to the reducedEG object, can use original object ptr
-  // or we are running MINIAOD->MINIAOD and we need to fetch the pat objects to reference
-  edm::Ptr<reco::Candidate> ptr(pho.originalObjectRef());
-
-  if(!ph_conf.tok_photon_src.isUninitialized()) {
-    auto key = phos_by_oop.find(pho.originalObjectRef().key());
-    if( key != phos_by_oop.end() ) {
-      ptr = key->second;
-    } else {
-      throw cms::Exception("BadPhotonKey")
-        << "Original object pointer with key = " << pho.originalObjectRef().key() << " not found in cache!";
-    }
-  }
+void EGExtraInfoModifierFromDB::modifyObject(pat::Electron& ele) const {
+  modifyObject(static_cast<reco::GsfElectron&>(ele));
+}
+
+void EGExtraInfoModifierFromDB::modifyObject(reco::Photon& pho) const {
+  // regression calculation needs no additional valuemaps
   
   std::array<float, 31> eval;
-  reco::SuperClusterRef sc = pho.superCluster();
-  edm::Ptr<reco::CaloCluster> theseed = sc->seed();
+  const reco::SuperClusterRef& the_sc = pho.superCluster();
+  const edm::Ptr<reco::CaloCluster>& theseed = the_sc->seed();
   
+  const int numberOfClusters =  the_sc->clusters().size();
+  const bool missing_clusters = !the_sc->clusters()[numberOfClusters-1].isAvailable();
+
+  if( missing_clusters ) return ; // do not apply corrections in case of missing info (slimmed MiniAOD electrons)
+
+  const double raw_energy = the_sc->rawEnergy(); 
+  const auto& ess = pho.showerShapeVariables();
+
   // SET INPUTS
-  eval[0]  = sc->rawEnergy();
-  //eval[1]  = sc->position().Eta();
-  //eval[2]  = sc->position().Phi();
+  eval[0]  = raw_energy;
+  //eval[1]  = the_sc->position().Eta();
```

Comment:
are these commented out lines needed anymore? (current values suggest completely different meaning for inds 1,2.